core/payment/src/api/allocations.rs

```diff
@@ -38,6 +39,71 @@ pub fn register_endpoints(scope: Scope) -> Scope {
         .route("/demandDecorations", get().to(get_demand_decorations))
 }
 
+fn validate_network(network: &str) -> Result<(), String> {
+    match network {
+        "mainnet" => Ok(()),
+        "rinkeby" => Err("Rinkeby is no longer supported".to_string()),
+        "goerli" => Ok(()),
+        "holesky" => Ok(()),
+        "polygon" => Ok(()),
+        "mumbai" => Ok(()),
+        _ => Err(format!("Invalid network name: {network}")),
+    }
+}
```

Comment:
i think you are duplicating a lot of code here. We have enum for network already, which could be just deserialized.